crawl4ai/async_crawler_strategy.py

```
@@ -900,12 +901,33 @@ async def handle_request_failed_capture(request):
                 await self.execute_hook("on_execution_ended", page, context=context, config=config, result=execution_result)
 
-            # Handle user simulation
+            # Handle user simulation with natural human-like behavior
             if config.simulate_user or config.magic:
-                await page.mouse.move(100, 100)
-                await page.mouse.down()
-                await page.mouse.up()
-                await page.keyboard.press("ArrowDown")
+                # Generate random start and end points for mouse movement
+                viewport = page.viewport_size
+                start_x = random.randint(50, viewport["width"] // 3)
```

Ruff (0.14.0)

Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)